### PR TITLE
CMS-37: Add park contacts collection

### PR DESCRIPTION
**Pull request by ayumi-oxd**
### Jira Ticket:
CMS-37
CMS-260

### Description:
- Add park contacts collection
- Add park operator contact collection
- Add a default contact

### CODE REVIEW

File: src/cms/src/components/contact/link.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        "laptop",
+        "messages"
+      ],
+      "required": true
+    },
+    "linkText": {
+      "type": "string",
+      "required": true
+    },
+    "linkUrl": {
```
Comment (molund) [inline src/cms/src/components/contact/link.json:24]: We should make a note to add help text for this field after deployment.

e.g. 
Email and phone URLs should NOT include as scheme (e.g. mailto/tel). If you leave this field blank, the linkText will be rendered as plain text with no link.

File: src/gatsby/src/components/park/contact.js [hunk 2]
```diff
@@ -1,11 +1,117 @@
 import React from "react"
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
 import HtmlContent from "./htmlContent"
 
-export default function Contact({ contact }) {
+// Helper function to get href prefix based on icon
```
Comment (molund) [inline src/gatsby/src/components/park/contact.js:5]: I'm not sure if it's intuitive that we are adding the scheme for tel/mailto and not http/https.  Maybe check with UX.

File: src/cms/src/api/park-contact/content-types/park-contact/lifecycles.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+"use strict";
+
+const updateName = async (data, where) => {
```
Comment (molund) [inline src/cms/src/api/park-contact/content-types/park-contact/lifecycles.js:3]: The name only gets updated when you update a contact, not when you add a new contact.  Is that consistent with how other collections work? I thought it was broken at first

File: src/gatsby/gatsby-config.js [hunk 4]
```diff
@@ -147,8 +149,11 @@ module.exports = {
                     populate: ["isActive", {
                       campingType: ["campingTypeNumber", "campingTypeCode", "isActive"]
                     }]
-                  },                  
+                  },
                 },
+                parkContacts: {
+                  populate: ["parkOperatorContact"]
```
Comment (molund) [inline src/gatsby/gatsby-config.js:155]: I'm pretty sure this isn't working right.  

parkContacts.contactInformation and parkOperatorContact.defaultContactInformation aren't being populated.

File: src/cms/src/api/park-contact/content-types/park-contact/lifecycles.js (hunk 3 above)
Comment (ayumi-oxd) [reply src/cms/src/api/park-contact/content-types/park-contact/lifecycles.js:3]: It works when you add a new park-contact. `name` will be automatically created when the publish button is clicked. It's the same function as park-activity and park-facility. Please let me know if it doesn't work on your end.

File: src/gatsby/gatsby-config.js (hunk 4 above)
Comment (ayumi-oxd) [reply src/gatsby/gatsby-config.js:155]: I couldn't recreate your issue. Did you check at the API endpoint after giving permission to these collections on Strapi? It works on my end through GraphQL. Please let me know how you faced the issue.
 
Screenshot 2024-07-29 at 11 16 41 AM
Screenshot 2024-07-29 at 11 16 51 AM

File: src/cms/src/api/park-contact/content-types/park-contact/lifecycles.js (hunk 3 above)
Comment (molund) [reply src/cms/src/api/park-contact/content-types/park-contact/lifecycles.js:3]: Sounds good. I didn't press publish

**Review by molund — APPROVED** [review]
This is approved, but please don't merge it until after we release the alpha branch to UAT

**Pull request merged**